lib/shared.js

```diff
@@ -1,109 +1,104 @@
+        let redisAddons = []
+        for (let i = 0; i < addons.length; i++) {
+          let addon = addons[i]
+          let service = addon.addon_service.name
 
-      if (service.indexOf(ADDON) === 0 && (!filter || matches(addon))) {
-        redisAddons.push(addon)
+          if (service.indexOf(ADDON) === 0 && (!filter || matches(addon))) {
+            redisAddons.push(addon)
+          }
+        }
+        return redisAddons
       }
-    }
-    return redisAddons
-  }
 
-  return onResponse
-}
+      return onResponse
+    },
 
-function * getRedisAddon (context, heroku, addonsList) {
-  addonsList = addonsList || heroku.get(`/apps/${context.app}/addons`)
+    async getRedisAddon (addonsList) {
+      addonsList = addonsList || heroku.get(`/apps/${context.app}/addons`)
 
-  let addonsFilter = makeAddonsFilter(context.args.database)
-  let addons = addonsFilter(yield addonsList)
+      let addonsFilter = this.makeAddonsFilter(context.args.database)
+      let addons = addonsFilter(await addonsList)
 
-  if (addons.length === 0) {
-    cli.exit(1, 'No Redis instances found.')
-  } else if (addons.length > 1) {
-    let names = addons.map(function (addon) { return addon.name })
-    cli.exit(1, `Please specify a single instance. Found: ${names.join(', ')}`)
-  }
+      if (addons.length === 0) {
+        cli.exit(1, 'No Redis instances found.')
+      } else if (addons.length > 1) {
+        let names = addons.map(function (addon) { return addon.name })
+        cli.exit(1, `Please specify a single instance. Found: ${names.join(', ')}`)
+      }
 
-  return addons[0]
-}
+      return addons[0]
+    },
 
-function * info (context, heroku) {
-  let addons = yield heroku.get(`/apps/${context.app}/addons`)
-  // filter out non-redis addons
-  addons = makeAddonsFilter(context.args.database)(addons)
-  // get info for each db
-  let databases = yield addons.map(function * (addon) {
-    return yield {
-      addon: addon,
-      redis: request(context, `/redis/v0/databases/${addon.name}`).catch(function (err) {
-        if (err.statusCode !== 404) {
-          throw (err)
+    async info () {
+      let addons = await heroku.get(`/apps/${context.app}/addons`)
+      // filter out non-redis addons
+      addons = this.makeAddonsFilter(context.args.database)(addons)
+      // get info for each db
+      let databases = await addons.map(addon => {
+        return {
+          addon: addon,
+          redis: this.request(`/redis/v0/databases/${addon.name}`).catch(function (err) {
+            if (err.statusCode !== 404) {
+              throw (err)
+            }
+            return null
+          })
         }
-        return null
       })
-    }
-  })
 
-  // print out the info of the addon and redis db info
-  for (let db of databases) {
-    if (db.redis === null) {
-      continue
-    }
+      // print out the info of the addon and redis db info
+      for (let db of databases) {
+        const redis = await db.redis
```

Comment:
Oh I see--you're `awai`ting just-in-time here instead, right? In which case you don't need the `await` above at all?

Reply:
Yeah you're right, I'll remove that extra one